Can't set action.auto_create_index to any value(too restrictive)

**Elasticsearch version**: 6.4.0

**Plugins installed**: []

**JVM version**: OpenJDK 1.8.0_181

**OS version**: Linux 3.16.0-5-amd64 #1 SMP Debian 3.16.51-3+deb8u1 (2018-01-08) x86_64 GNU/Linux

**Description of the problem including expected versus actual behavior**:
When I try to set `action.auto_create_index: false` I get the following error when starting Elasticsearch:

`Caused by: java.lang.IllegalArgumentException: the [action.auto_create_index] setting value [false] is too restrictive. disable [action.auto_create_index] or set it to [.watches, .triggered_watches, .watcher-history-*]`

After trying some different values(`-*` and `+foobar,-*`) I decided to try setting it to exactly what the error message recommended: `action.auto_create_index: watches, .triggered_watches, .watcher-history-*` which is when I got the amazing error:

`Caused by: java.lang.IllegalArgumentException: the [action.auto_create_index] setting value [.watches, .triggered_watches, .watcher-history-*] is too restrictive. disable [action.auto_create_index] or set it to [.watches, .triggered_watches, .watcher-history-*]`

The initial issue still isn't solved(still can't disable automatic index creation), but this seems like a documentation/errormessage bug at the very least.

**Steps to reproduce**:

 1. Install Elasticsearch 6.4.0
 2. Put `action.auto_create_index: false` in elasticsearch.yml
 3. (Re)start Elasticsearch

--- a/x-pack/plugin/watcher/src/main/java/org/elasticsearch/xpack/watcher/Watcher.java
+++ b/x-pack/plugin/watcher/src/main/java/org/elasticsearch/xpack/watcher/Watcher.java
@@ -535,7 +535,7 @@ public class Watcher extends Plugin implements ActionPlugin, ScriptPlugin, Reloa
 
         String errorMessage = LoggerMessageFormat.format("the [action.auto_create_index] setting value [{}] is too" +
                 " restrictive. disable [action.auto_create_index] or set it to " +
-                "[{}, {}, {}*]", (Object) value, Watch.INDEX, TriggeredWatchStoreField.INDEX_NAME, HistoryStoreField.INDEX_PREFIX);
+                "[{},{},{}*]", (Object) value, Watch.INDEX, TriggeredWatchStoreField.INDEX_NAME, HistoryStoreField.INDEX_PREFIX);
         if (Booleans.isFalse(value)) {
             throw new IllegalArgumentException(errorMessage);
         }

--- a/x-pack/plugin/watcher/src/test/java/org/elasticsearch/xpack/watcher/WatcherPluginTests.java
+++ b/x-pack/plugin/watcher/src/test/java/org/elasticsearch/xpack/watcher/WatcherPluginTests.java
@@ -37,7 +37,7 @@ public class WatcherPluginTests extends ESTestCase {
 
         IllegalArgumentException exception = expectThrows(IllegalArgumentException.class,
                 () -> Watcher.validAutoCreateIndex(Settings.builder().put("action.auto_create_index", false).build(), logger));
-        assertThat(exception.getMessage(), containsString("[.watches, .triggered_watches, .watcher-history-*]"));
+        assertThat(exception.getMessage(), containsString("[.watches,.triggered_watches,.watcher-history-*]"));
 
         Watcher.validAutoCreateIndex(Settings.builder().put("action.auto_create_index",
                 ".watches,.triggered_watches,.watcher-history*").build(), logger);
@@ -46,16 +46,16 @@ public class WatcherPluginTests extends ESTestCase {
 
         exception = expectThrows(IllegalArgumentException.class,
                 () -> Watcher.validAutoCreateIndex(Settings.builder().put("action.auto_create_index", ".watches").build(), logger));
-        assertThat(exception.getMessage(), containsString("[.watches, .triggered_watches, .watcher-history-*]"));
+        assertThat(exception.getMessage(), containsString("[.watches,.triggered_watches,.watcher-history-*]"));
 
         exception = expectThrows(IllegalArgumentException.class,
                 () -> Watcher.validAutoCreateIndex(Settings.builder().put("action.auto_create_index", ".triggered_watch").build(), logger));
-        assertThat(exception.getMessage(), containsString("[.watches, .triggered_watches, .watcher-history-*]"));
+        assertThat(exception.getMessage(), containsString("[.watches,.triggered_watches,.watcher-history-*]"));
 
         exception = expectThrows(IllegalArgumentException.class,
                 () -> Watcher.validAutoCreateIndex(Settings.builder().put("action.auto_create_index", ".watcher-history-*").build(),
                         logger));
-        assertThat(exception.getMessage(), containsString("[.watches, .triggered_watches, .watcher-history-*]"));
+        assertThat(exception.getMessage(), containsString("[.watches,.triggered_watches,.watcher-history-*]"));
     }
 
     public void testWatcherDisabledTests() throws Exception {

